addons/mail/controllers/discuss/gif.py

```diff
@@ -0,0 +1,71 @@
+            return False
+        return response
+
+    @route("/discuss/gif/search", type="json", auth="public", cors="*")
+    def search(self, search_term, locale="en", country="US", pos=False):
+        config = request.env["ir.config_parameter"].sudo()
+        api_key = config.get_param("discuss.tenor_api_key")
+        content_filter = config.get_param("discuss.tenor_content_filter")
+        ckey = request.env.cr.dbname
+        tenor_url = config.get_param("discuss.tenor_url")
+        tenor_gif_limit = config.get_param("discuss.tenor_gif_limit")
+        url = f"{tenor_url}/v2/search?q={search_term}&key={api_key}&client_key={ckey}&limit={tenor_gif_limit}&contentfilter={content_filter}&locale={locale}&country={country}&media_filter=tinygif"
+        if pos:
+            url = f"{url}&pos={pos}"
+        response = self.request_gifs(url)
+        if response:
+            return json.loads(response.content)
+
+    @route("/discuss/gif/categories", type="json", auth="public", cors="*")
+    def categorie(self, locale="en", country="US"):
+        config = request.env["ir.config_parameter"].sudo()
+        api_key = config.get_param("discuss.tenor_api_key")
+        content_filter = config.get_param("discuss.tenor_content_filter")
+        ckey = request.env.cr.dbname
+        tenor_url = config.get_param("discuss.tenor_url")
+        url = f"{tenor_url}/v2/categories?key={api_key}&client_key={ckey}&contentfilter={content_filter}&locale={locale}&country={country}"
+        response = self.request_gifs(url)
+        if response:
+            return json.loads(response.content)
+
+    @route("/discuss/gif/add_favorite", type="json", auth="user")
+    def add_favorite(self, tenor_gif_id):
+        request.env["discuss.gif.favorite"].create({"tenor_gif_id": tenor_gif_id})
+
+    def _gif_posts(self, ids):
+        config = request.env["ir.config_parameter"].sudo()
+        ckey = request.env.cr.dbname
+        api_key = config.get_param("discuss.tenor_api_key")
+        tenor_url = config.get_param("discuss.tenor_url")
+        url = f"{tenor_url}/v2/posts?key={api_key}&client_key={ckey}&ids={','.join(ids)}&media_filter=tinygif"
+        response = self.request_gifs(url)
+        if response:
+            return json.loads(response.content)["results"]
+
+    @route("/discuss/gif/favorites", type="json", auth="user")
+    def get_favorites(self, offset=0):
+        tenor_gif_ids = request.env["discuss.gif.favorite"].search([("create_uid", "=", request.env.user.id)], limit=20, offset=offset)
+        return {
+            "results": self._gif_posts(tenor_gif_ids.mapped("tenor_gif_id")) or [],
+            "offset": offset + 20
+        }
+
+    @route("/discuss/gif/remove_favorite", type="json", auth="user")
+    def remove_favorite(self, tenor_gif_id):
+        return request.env["discuss.gif.favorite"].search([("create_uid", "=", request.env.user.id), ("tenor_gif_id", "=", tenor_gif_id)]).unlink()
```

Comment:
This is not the best way to delete them since I actually don't have the database favorite id.
Since I ask tenor for the favorite data, I'll have to reconcile the database id with the tenor gif id, witch is useless since I can just delete them by tenor_gif_id.